### ISSUE TITLE
Rule Change: no-plusplus rule is too restrictive

### ISSUE THREAD

**Issue by michael-freidgeim-webjet** [eslint/eslint#15377]
### What rule do you want to change?

no-plusplus

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Other

### Example code

```js
The reason of ["disallow the unary operators ++ and -- (no-plusplus)" ](https://eslint.org/docs/rules/no-plusplus)
 is ambiguous code if unary ++ and -- operators are subject to automatic semicolon insertion.
i.e. when semicolon is missing and operator  is not directly follows the variable.

However current implementation disallow valid usage with ; and operator next to variable, e.g.
bar--;
for (i = 0; i < l; j = i++) 

It will be good , if rule will warn only ambiguous conditions and do not report normal usage of the unary operators.
```

### What does the rule currently do for this code?

bar--;
for (i = 0; i < l; j = i++) 
reports warning

### What will the rule do after it's changed?

For
bar--;
for (i = 0; i < l; j = i++) 
Do not report warning.

For 
i
++
j
report warning.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.
